x-pack/plugins/search_playground/public/hooks/use_search_preview.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useFormContext } from 'react-hook-form';
+import { SearchHit } from '@elastic/elasticsearch/lib/api/types';
+import useSWR from 'swr';
+import { APIRoutes, ChatFormFields, Pagination } from '../types';
+import { useKibana } from './use_kibana';
+import { DEFAULT_PAGINATION } from '../../common';
+
+export interface FetchSearchResultsArgs {
+  query: string;
+  pagination: Pagination;
+}
+
+interface UseSearchPreviewData {
+  results: SearchHit[];
+  pagination: Pagination;
+  isInitialState: boolean;
+}
+
+export interface UseSearchPreviewResponse {
+  fetchSearchResults: (args: FetchSearchResultsArgs) => Promise<void>;
+  data: UseSearchPreviewData;
+}
+
+const DEFAULT_SEARCH_PREVIEW_DATA: UseSearchPreviewData = {
+  results: [],
+  pagination: DEFAULT_PAGINATION,
+  isInitialState: true,
+};
+
+export const useSearchPreview = (): UseSearchPreviewResponse => {
+  const { getValues } = useFormContext();
+  const { services } = useKibana();
+  const { http } = services;
+
+  const { data, mutate } = useSWR<UseSearchPreviewData>('search-preview-results', null, {
```

Comment:
I would suggest you use tanstack useQuery here instead. I know useSWR is used for chat but there was a couple of challenges using useQuery for streaming.

Reply:
i am currently doing this, tomorrow it will be updated. 
Small rant: "Out of the box, TanStack Query is configured with aggressive but sane defaults" statement is not correct. :)